File: Python/2021/Day02.py
import InputHandler

def Run(filepath):
    directions = InputHandler.ToStringList(filepath, '\n')

    print('Day 2-1: ' + Part1(directions)) #The result should be 1989014
    print('Day 2-2: ' + Part2(directions)) #The result should be 2006917119
    
def Part1(directions):
    horizontal = 0
    depth = 0

    for instruction in directions:
        #Split the full instruction into direction and distance
        direction = instruction.split(' ')[0]
        distance = instruction.split(' ')[1]
        if direction == 'forward':
            horizontal += int(distance)
        elif direction == 'down':
            depth += int(distance)
        elif direction == 'up':
            depth -= int(distance)

    answer = horizontal * depth

    return str(answer)

def Part2(directions):    
    horizontal = 0
    depth = 0
    aim = 0

    for instruction in directions:
        #Split the full instruction into direction and distance
        direction = instruction.split(' ')[0]
        distance = instruction.split(' ')[1]
        if direction == 'forward':
            horizontal += int(distance)
            depth += aim * int(distance)
        elif direction == 'down':
            aim += int(distance)
        elif direction == 'up':
            aim -= int(distance)

    answer = horizontal * depth

    return str(answer)

File: CSharp/App/2021/Day02.cs
using AdventOfCode;

namespace AOC2021;
public class Day02 : Day{
    public override long Part1(string filepath){
        return PlotCourse(Inputs.ListLines(filepath), false);
    }
    public override long Part2(string filepath){
        return PlotCourse(Inputs.ListLines(filepath), true);
    }
    static int PlotCourse(List<string> directions, bool useAim){
        int horizontal = 0, depth = 0, aim = 0;
        foreach (string direction in directions){
            //Split the full instruction into a tuple containing direction and distance
            string[] directionSplit = direction.Split(" ");
            var (heading, distance) = (directionSplit[0], Convert.ToInt32(directionSplit[1]));
            switch (heading){
                case "forward":
                    horizontal += distance;
                    if (useAim){ depth += aim * distance; }
                    break;
                case "down":
                    if (useAim){ aim += distance; }
                    else { depth += distance; }
                    break;
                case "up":
                    if (useAim){ aim -= distance; }
                    else { depth -= distance; }
                    break;
            }
        }
        return horizontal * depth;
    }
}